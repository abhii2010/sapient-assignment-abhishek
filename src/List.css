.item {
  background-color: white;
  margin: 0 0 10px 10px;
  padding: 10px;
  max-width: 270px;
}

.list {
  display: grid;
  grid-template-columns: auto auto auto auto;
}

.detail-title {
  display: block;
  font-weight: 700;
}

.detail-head{
  display: block;
  color: #292980;
  font-weight: 700;
}

.sub-section {
  color: #292980;
}

.img-class {
  background-color: lightgray;
  margin-bottom: 10px;
}

