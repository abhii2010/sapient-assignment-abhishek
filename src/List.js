import React from "react";
import './List.css';

const List = (props) => {
  const  { records } = props;

  return (<div className="list">
    {records.map(record => (
      <div key={record.mission_name} className="item">
        <img className="img-class" src={record.links.mission_patch_small} alt={record.mission_name} />
        <div className="detail">
          <span className="detail-head">{record.mission_name}</span>
          <span className="detail-title">Mission Ids</span>
          <ul>
            {
              record.mission_id.map(id => (
                <li key={id} >{id}</li>
              ))
            }
          </ul>
          <span className="detail-title">Launch year: <label className="sub-section">{record.launch_year}</label></span>
          <span className="detail-title">Successful launch: <label className="sub-section">{record.launch_success && record.launch_success.toString()}</label></span>
          <span className="detail-title">Successful landing: <label className="sub-section">{record.landing_success}</label></span>
        </div>
      </div>
    ))}
  </div>);
};

export default List;
