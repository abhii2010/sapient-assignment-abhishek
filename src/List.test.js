import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

const mockRecords = [
  {
    launch_success: true,
    landing_success: true,
    launch_year: '2006',
    mission_name: 'test',
    links: {
      mission_patch_small: ''
    },
    mission_id: []
  }
];

let wrapped = shallow(<List records={mockRecords}/>);
describe('List', () => {
  test('snapshot renders', () => {
    expect(wrapped).toMatchSnapshot();
  });
});