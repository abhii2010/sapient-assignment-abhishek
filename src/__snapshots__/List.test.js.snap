// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List snapshot renders 1`] = `
<div
  className="list"
>
  <div
    className="item"
    key="test"
  >
    <img
      alt="test"
      className="img-class"
      src=""
    />
    <div
      className="detail"
    >
      <span
        className="detail-head"
      >
        test
      </span>
      <span
        className="detail-title"
      >
        Mission Ids
      </span>
      <ul />
      <span
        className="detail-title"
      >
        Launch year: 
        <label
          className="sub-section"
        >
          2006
        </label>
      </span>
      <span
        className="detail-title"
      >
        Successful launch: 
        <label
          className="sub-section"
        >
          true
        </label>
      </span>
      <span
        className="detail-title"
      >
        Successful landing: 
        <label
          className="sub-section"
        >
          <Component />
        </label>
      </span>
    </div>
  </div>
</div>
`;
